Sources/NIO/Socket.swift

```diff
@@ -170,6 +170,33 @@ typealias IOVector = iovec
         }
     }
 
+    func recvmsg(pointer: UnsafeMutableRawBufferPointer, storage: inout sockaddr_storage, storageLen: inout socklen_t, controlBytes: inout UnsafeMutableRawBufferPointer) throws -> IOResult<Int> {
+        var vec = iovec(iov_base: pointer.baseAddress, iov_len: pointer.count)
+
+        return try withUnsafeMutablePointer(to: &vec) { vecPtr in
+            return try storage.withMutableSockAddr { (sockaddrPtr, _) in
+                var messageHeader = msghdr(msg_name: sockaddrPtr,
+                                           msg_namelen: storageLen,
+                                           msg_iov: vecPtr,
+                                           msg_iovlen: 1,
+                                           msg_control: controlBytes.baseAddress,
+                                           msg_controllen: .init(controlBytes.count),
```

Comment:
thanks UNIX for providing such fantastic APIs :)

Reply:
It's bloody annoying. `msg_controllen` is `socklen_t` on Darwin and `ssize_t` on Linux.

Reply:
@Lukasa oh sorry! Mis-reviewed then. My opinion is `&` is (obviously) fine and good for `inout` but bad for `Unsafe*Pointer`. Fairly sure we share that opinion anyway :D. With Swift had a sigil that _only_ worked on `inout`s so that I won't misreview like this in the future :)